Community/AssetManagement/lumeta_workflow_page.py

```diff
@@ -0,0 +1,311 @@
+            # Add ip address
+            ip_obj = assign_ip(network_data, ip, mac, family)
+            added_ip += 1
+
+    # If no  block exists and add_network is set to true, create Block with /32, create Network with /32 and then
+    # create ip with /32
+    except PortalException as e:
+        # No block address containing input ip address exists. Check the flag and create one
+        if add_network:
+            try:
+                # Add Block, then network and finally add ip
+                # Below line is returning BAMException - IPv4 Blocks cannot be in size of /31 and /32
+                # So, at this point, if there is no container, do not add ip address
+                # config.add_ip4_block_by_cidr(ip)
+                if blk_data is None:
+                    # add /30 for addressblock
+                    block_network = ipaddress.ip_network(ip + '/30', strict=False)
+                    config.add_ip4_block_by_cidr(block_network.exploded)
+                    blk_data = config.get_ip_range_by_ip('IP4Block', ip)
+
+                if blk_data is not None:
+                    # create network  in block
+                    blk_data.add_ip4_network(ip + '/32')
+
+                    # create ip under above created network
+                    network_data = config.get_ip_range_by_ip('IP4Network', ip)
+                    if network_data is not None:
+                        # Add ip address
+                        ip_obj = assign_ip(network_data, ip, mac, family)
+                        added_ip += 1
+            except APIException as ex:
+                if "Duplicate" in ex.get_message():
+                    dup_ip += 1
+                # else:
+                # Seeing intermittent error while adding address block, so had to stop logging error
+                # app.loggererror('add_ip: ' + ex.message)
+    except APIException as ex:
+        # when ip address already exists, it returns BAMException with message 'Server raised fault: "Duplicate of another item"'
+        # "Duplicate" in ex.get_message()
+        if "Duplicate" in ex.get_message():
+            dup_ip += 1
+        else:
+            # TODO - how to log info message and not error?
+            app.loggererror('add_ip: ' + ex.get_message())
+    return (added_ip, dup_ip, ip_obj)
+
+
+def assign_ip(network_data, ip, mac, family):
+    if mac is not '':
+        if family == '4':
+            ip = network_data.assign_ip4_address(ip, mac, '', 'MAKE_DHCP_RESERVED')
+        else:
+            ip = network_data.assign_ip6_address(ip, mac, '', 'MAKE_DHCP_RESERVED')
+    else:
+        if family == '4':
+            ip = network_data.assign_ip4_address(ip, '', '', 'MAKE_STATIC')
+        else:
+            ip = network_data.assign_ip6_address(ip, '', '', 'MAKE_STATIC')
+    return ip
+
+
+def add_tag(ip):
+    tag_group = None
+    tag = None
+    try:
+        tag_group = api.get_tag_group_by_name("Lumeta")
```

Comment:
The tag group should be called out as a dependency in the README, otherwise, a user wouldn't know what to call the tag group without looking into the code. Ideally, magic words belong in a configuration or as a global.